src/fluka/fluka_funcs.cpp

```diff
@@ -0,0 +1,1104 @@
+void jomiwr(int& nge, const int& lin, const int& lou, int& flukaReg) {
+
+  if (debug) {
+    std::cout << "================== JOMIWR =================" << std::endl;
+  }
+
+  //Original comment:  returns number of volumes
+  unsigned int numVol = DAG->num_entities(3);
+  flukaReg = numVol;
+
+  if (debug) {
+    std::cout << "Number of volumes: " << flukaReg << std::endl;
+    std::cout << "================== Out of JOMIWR =================" << std::endl;
+  }
+
+  return;
+}
+
+/* returns the approved step from pS along pV */
+void g_step(double& pSx,
+            double& pSy,
+            double& pSz,
+            double* pV,
+            int& oldReg,         // pass through
+            const int& oldLttc,  // ignore
+            double& propStep,    // .
+            int& nascFlag,       // .
+            double& retStep,     // reset in this method
+            int& newReg,         // return from callee
+            double& saf,         // safety
+            int& newLttc,        // .
+            int& LttcFlag,       // .
+            double* sLt,         // .
+            int* jrLt) {         // .
+  double safety; // safety parameter
+
+  if (debug) {
+    std::cout << "============= G_STEP	 ==============" << std::endl;
+    std::cout << "Position " << pSx << " " << pSy << " " << pSz << std::endl;
+    std::cout << "Direction vector " << pV[0] << " " << pV[1] << " " << pV[2] << std::endl;
+    std::cout << "Oldreg = " << oldReg << std::endl;
+    std::cout << "PropStep = " << propStep << std::endl;
+  }
+
+  double point[3] = {pSx, pSy, pSz};
+  double dir[3]   = {pV[0], pV[1], pV[2]};
+
+  g_fire(oldReg, point, dir, propStep, retStep, saf, newReg); // fire a ray
+
+  if (debug) {
+    std::cout << " ret = " << retStep;
+    std::cout << " new cel = " << newReg << std::endl;
+    std::cout << "saf = " << saf << std::endl;
+    std::cout << std::setw(20) << std::scientific;
+    std::cout << "newReg = " << newReg << " retStep = " << retStep << std::endl;
+  }
+
+  return;
+}
+
+/* function to determine the particles next volume & step length */
+void g_fire(int& oldRegion, double point[], double dir[], double& propStep,
+            double& retStep, double& safety,  int& newRegion) {
+  moab::EntityHandle vol = DAG->entity_by_index(3, oldRegion); // get eh of current region
+  moab::EntityHandle next_surf; // next surf we hit
+  double next_surf_dist;
+  moab::EntityHandle newvol = 0;
+
+  if (debug)
+    print_state(state);
+
+  // direction changed reset history, may not be robust
+  if (dir[0] != state.old_direction[0] || dir[1] != state.old_direction[1] || dir[2] != state.old_direction[2]) {
+    if (point[0] != state.old_position[0] || point[1] != state.old_position[1] || point[2] != state.old_position[2]) {
+      // if point and direction has changed, we should reset everything
+      reset_state(state);
+    } else {
+      // direction has changed reset ray history
+      state.history.reset();
+    }
+  }
+  /*
```

Comment:
Readability: Remove commented out code, or document under what conditions it can be removed.
Here and throughout the file.